### PR TITLE
fix: Improved print format of Statement of account report

### PR DESCRIPTION
**Pull request by meshramsaroj**
**Task Link:** https://app.asana.com/0/1191634438432931/1191716019962961/f

**Description:** Beautified print format of Statement of account report using CSS style.

**Screenshot:**

- Before

before_SOA

- After

updated-soa

### CODE REVIEW

File: erpnext/accounts/report/statement_of_account/statement_of_account.html [hunk 1]
```diff
@@ -1,28 +1,103 @@
...
-</h5>
-<hr>
+<style>
+.print-format.landscape {
+	margin: 100px;
+}
+
+.print-format td,.print-format th{
+	padding: 12px !important ;
+}
```
Comment (nehasacher143) [inline erpnext/accounts/report/statement_of_account/statement_of_account.html]: @meshramsaroj Please add a space Bracket and end of style rule
Do it for all other rules

File: erpnext/accounts/report/statement_of_account/statement_of_account.html [hunk 2]
```diff
@@ -1,28 +1,103 @@
...
+	float:right;
+	width: 40%;
+}
+.address .row p ,.statement-date .row label,.value{
+	color: #948a8a;
+	font-size: 15px;
+}
+
+</style>
+<div class=row>
```
Comment (nehasacher143) [inline erpnext/accounts/report/statement_of_account/statement_of_account.html]: Double quotes to the class name

File: erpnext/accounts/report/statement_of_account/statement_of_account.html [hunk 3]
```diff
@@ -1,28 +1,103 @@
...
+	<div class="col-sm-6 address">
+		<div class="row">
+			<p> As per company letter head address</p>
+		</div>
+	</div>
+	<div class="col-sm-6 statement-date ">
```
Comment (nehasacher143) [inline erpnext/accounts/report/statement_of_account/statement_of_account.html]: Remove an extra space here

File: erpnext/accounts/report/statement_of_account/statement_of_account.html [hunk 4]
```diff
@@ -1,28 +1,103 @@
...
+			</div>
+			<div class="col-xs-7  value text-right">
+				{%= frappe.datetime.str_to_user(filters.from_date) %}  {%= __("to") %}  {%= frappe.datetime.str_to_user(filters.to_date) %}
+			</div>
+		</div>
+	</div>
+</div>
+
+<div class="customer-info">
+	<h2 >{%= __("Statement of Account") %}</h2>
```
Comment (nehasacher143) [inline erpnext/accounts/report/statement_of_account/statement_of_account.html]: Why there are spaces before '>' in entire code. Please rectify

File: erpnext/accounts/report/statement_of_account/statement_of_account.html [hunk 5]
```diff
@@ -35,28 +110,25 @@ <h5 class="text-center">
...
-				<td style="text-align: right">
+				<td class="text-right" >
 					{%= format_currency(data[i].balance, filters.presentation_currency) %}
 				</td>
 			</tr>
 		{% } %}
 	</tbody>
 </table>
-<p class="text-right text-muted">Printed On {%= frappe.datetime.str_to_user(frappe.datetime.get_datetime_as_string()) %}</p>
+<div class="text-right text-muted">Printed On {%= frappe.datetime.str_to_user(frappe.datetime.get_datetime_as_string()) %}</div>
```
Comment (nehasacher143) [inline erpnext/accounts/report/statement_of_account/statement_of_account.html]: Can you please explain why this change

**Comment by nehasacher143** [comment]
Hey @Alchez Code looks good to me as well as the design
Please merge this if this is okay to you.

File: erpnext/accounts/report/statement_of_account/statement_of_account.html [hunk 6]
```diff
@@ -1,26 +1,111 @@
...
+
+	<p>
+		{% if (filters.tax_id) { %}
+		{%= __("Tax Id: ")%} {%= filters.tax_id %}
+		{% } %}
+	</p>
+
+	<!-- <p>
+		{% frappe.db.get_value("Dynamic Link", {"link_doctype": "Customer", "link_name": filters.party[0]},"parent") %}
+	</p> -->
```
Comment (Alchez) [inline erpnext/accounts/report/statement_of_account/statement_of_account.html]: We still need to figure out how to fetch the addresses, I'll have to look at this.

Comment (nehasacher143) [reply erpnext/accounts/report/statement_of_account/statement_of_account.html]: @Alchez Can we not merge it till then?

Comment (Alchez) [reply erpnext/accounts/report/statement_of_account/statement_of_account.html]: No point merging it, really, without the entire format done

**Comment by Alchez** [comment]
@meshramsaroj, resolve the conflicts here

File: erpnext/accounts/report/statement_of_account/statement_of_account.js [hunk 7]
```diff
@@ -99,7 +105,27 @@ frappe.query_reports["Statement of Account"] = {
...
+	],
 };
 
 erpnext.utils.add_dimensions('Statement of Account', 15)
+
+function get_addresses(report) {
+	let filters = report.get_filter_values();
+
+	if (!filters.company || !filters.party_type || !filters.party[0]) {
+		return
```
Comment (Alchez) [inline erpnext/accounts/report/statement_of_account/statement_of_account.js]:
```suggestion
		return;
```

File: erpnext/accounts/report/statement_of_account/statement_of_account.js [hunk 8]
```diff
@@ -99,7 +105,27 @@ frappe.query_reports["Statement of Account"] = {
...
+	}
+
+	frappe.call({
+		method: "erpnext.accounts.report.statement_of_account.statement_of_account.get_addresses",
+		args: {
+			company: filters.company,
+			party_type: filters.party_type,
+			party: filters.party[0]
```
Comment (Alchez) [inline erpnext/accounts/report/statement_of_account/statement_of_account.js]: Talk to @vishdha and Arundhati about this, maybe we don't want to allow multiple parties in that field.

**Review by Alchez — CHANGES_REQUESTED** [review]
@meshramsaroj, the address logic is looking good. You just need to add it to the HTML now.

**Review by Alchez — APPROVED** [review]

**Pull request merged**